Lexicon v2

papajohn opened the pull request.

papajohn committed:
TODOs and ranking sketch

Commit:
adds functionality for scoring/ranking of meanings through a TM

Commit:
changes to LexiconProtocol member access for subclassing

Commit:
make logger protected to access in subclasses

Commit:
more protected/public changes for subclassing

papajohn commented:
I read through, but didn't have any major questions. I'll probably just tweak a little formatting and merge.

ahaas commented:
Cool! If there's any tweaks to the API let me know so I can update my usage.

On Tue, Nov 10, 2015 at 8:31 AM, John DeNero notifications@github.com
wrote:

> I read through, but didn't have any major questions. I'll probably just
> tweak a little formatting and merge.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/oscii-lab/lex/pull/11#issuecomment-155477398.

papajohn commented on src/main/java/org/oscii/concordance/RemoteAlignedCorpus.java:
```diff
@@ -67,9 +70,36 @@ public void read(String path, String sourceLanguage, String targetLanguage, int
+    double score = 0.0;
+
+    public PhrasalRule(String sourcePhrase, String targetPhrase, double score) {
+      if (sourcePhrase != null && targetPhrase != null) {
```
Is it possible that `sourcePhrase` is null but `targetPhrase` is not? It seems like the `Collections.emptyList()` substitutions should happen independently for source and target. I'm going to make that change and push — please comment before I merge.

sahasan commented:
That change should be fine. `PhrasalRule`s are created in `DecoderClientCorpus` from `RuleQueryResponse`s, and I think it is not possible that one of them is `null` and the other isn't. But better be on the safe side.

papajohn commented on src/main/java/org/oscii/concordance/RemoteAlignedCorpus.java:
```diff
@@ -67,9 +70,36 @@ public void read(String path, String sourceLanguage, String targetLanguage, int
+    @Override
+    public int compare(PhrasalRule x, PhrasalRule y) {
+      if (x != null && y != null) {
+        return Double.compare(y.score, x.score);
```
That's least-to-greatest. Is that what we want?

sahasan replied:
I think it's the other way around since x and y are switched in the compare call.

papajohn replied:
Aha, I see now. Thanks.

papajohn committed:
clean up

papajohn committed:
remove CSV parser

papajohn committed:
move PhrasalRule

papajohn commented:
Aside from the comments above, looks good to me! Merge at your leisure.

Commit:
adds query ranker and additional minor tweaks

- possibility to filter identity translations on read
- uses exact case filtering for query against prefix in extensions
- ranker for reordering query extensions
